source/common/http/filter_manager.cc

```diff
@@ -182,13 +182,15 @@ bool ActiveStreamFilterBase::commonHandleAfterDataCallback(FilterDataStatus stat
         status == FilterDataStatus::StopIterationAndWatermark) {
       buffer_was_streaming = status == FilterDataStatus::StopIterationAndWatermark;
       commonHandleBufferData(provided_data);
-    } else if (complete() && !hasTrailers() && !bufferedData()) {
+    } else if (complete() && !hasTrailers() && !bufferedData() && !parent_.state_.destroyed_) {
       // If this filter is doing StopIterationNoBuffer and this stream is terminated with a zero
       // byte data frame, we need to create an empty buffer to make sure that when commonContinue
       // is called, the pipeline resumes with an empty data frame with end_stream = true
       ASSERT(end_stream_);
       bufferedData() = createBuffer();
     }
+    // If the stream is destroyed, no need to handle the data buffer or trailers.
+    // This can occur if the filter calls sendLocalReply.
```

Comment:
Sorry to keep going back and forth on this, but can you please put this closer to the logic? (This code is confusing enough as it is) You can reformat the if statement and put this comment before the && condition or whatever you like. Thanks.

/wait

Reply:
No worries. I didn't think placing a comment inside an if condition was acceptable formatting, but it makes sense 👍